apiserver/v1/types_deprecatedapirequest.go

```diff
@@ -0,0 +1,110 @@
+// Package v1 is an api version in the apiserver.openshift.io group
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope="Cluster"
+// +kubebuilder:subresource:status
+// +genclient:nonNamespaced
+
+// DeprecatedAPIRequest tracts requests made to a deprecated API. The instance name should
+// be of the form `resource.version.group`, matching the deprecated resource.
+type DeprecatedAPIRequest struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Spec defines the characteristics of the resource.
+	// +kubebuilder:validation:Required
+	// +required
+	Spec DeprecatedAPIRequestSpec `json:"spec"`
+
+	// Status contains the observed state of the resource.
+	Status DeprecatedAPIRequestStatus `json:"status,omitempty"`
+}
+
+type DeprecatedAPIRequestSpec struct {
+	// RemovedRelease is when the API will be removed.
+	// +kubebuilder:validation:MaxLength=64
+	// +required
+	RemovedRelease string `json:"removedRelease"`
+}
+
+// +k8s:deepcopy-gen=true
+type DeprecatedAPIRequestStatus struct {
+
+	// Conditions contains details of the current status of this API Resource.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	Conditions []metav1.Condition `json:"conditions"`
+
+	// RequestsLastHour contains request history for the current hour. This is porcelain to make the API
+	// easier to read by humans seeing if they addressed a problem.
+	RequestsLastHour RequestLog `json:"requestsLastHour"`
```

Comment:
is this rolling? I.e. is it really the last 60min or at 3:47 it's everything from 3 to 4 ?

Reply:
It will be for the current "hour" , so from 3 to 4 in your example. This would match what we find in the `RequestsLast24h` slice.

Reply:
Maybe add "This is set back to zero every full hour."

Reply:
Updated,